### PR TITLE
Implement transaction_list

### LINKED ISSUE #131: [CONTRACT] Implement `transaction_list` Functionality in `AccountData` Component
### **Description**

Enhance the `AccountData` component of the account contract to support a transaction listing functionality. This function should allow fetching a paginated list of transactions with optional `start` and `limit` parameters. It should raise appropriate errors for invalid query ranges.

### **Function Signature**

```cairo
fn transaction_list(self: @TContractState, start: Option<u64>, limit: Option<u64>) -> Array<Transaction>;
```

### **Behavior**

* Returns a list of transactions starting from the `start` index (if provided), up to a maximum of `limit` transactions (if provided).
* If `start` is `None`, it defaults to `0`.
* If `limit` is `None`, it returns all available transactions from the `start` index onward.
* Raises an error if:

  * `start` is greater than or equal to the current number of transactions.
  * `start + limit` exceeds the number of available transactions.

### **Proposed Steps**

1. Add a new method `transaction_list` to the `AccountData` component.
2. Retrieve transactions from an internal mapping or array, respecting the pagination parameters.
3. Validate the `start` index:

   * If `start` >= `tx_count`, raise `TransactionIndexOutOfRange`.
4. Validate `start + limit` does not exceed the current transaction count:

   * If it does, raise `TransactionLimitOutOfRange`.
5. Return an array of transactions.
6. Write unit tests covering:

   * Normal pagination
   * Edge cases with `None` values
   * Invalid `start` or `start + limit` values

### **Acceptance Criteria**

* [ ] `transaction_list` method is implemented in `AccountData` with the described behavior.
* [ ] Errors are raised with meaningful messages or codes for out-of-range queries.
* [ ] All parameters are properly validated.
* [ ] Unit tests cover both happy path and error cases.
* [ ] Documentation/comments explain the behavior of `transaction_list`.

### PR DESCRIPTION
**Pull request by hakymulla**
## Description 📝
Implement transaction_list Functionality in AccountData Component 

## Related Issues 🔗
Closes #131 

## Changes Made 🚀
- [x] ✨ Feature Implementation 
- [ ] 🐛 Bug Fix 
- [ ] 📚 Documentation Update 
- [ ] 🔨 Refactoring 
- [ ] ❓ Others (Specify) 

## Checklist ✅
- [x] 🛠 I have tested these changes. 
- [x] 📖 I have updated the documentation (if applicable). 
- [x] 🎨 This PR follows the project's coding style. 
- [x] 🧪 I have added necessary tests (if applicable). 

## Summary by CodeRabbit

* **New Features**
  * Added the ability to retrieve a paginated list of transactions from an account, with optional start index and limit parameters.
* **Bug Fixes**
  * Introduced specific error messages for cases when the transaction index or limit is out of range.
* **Tests**
  * Added extensive tests covering transaction list retrieval, pagination, boundary conditions, and error scenarios.

### CODE REVIEW

File: src/account_data.cairo [hunk 1]
```diff
@@ -487,6 +487,43 @@ pub mod AccountData {
...
+        ) -> Array<Transaction> {
+            let transaction_count = self.tx_count.read();
+
+            if transaction_count == 0 {
+                return ArrayTrait::new();
+            }
+
+            let start_idx: u256 = match start {
+                Option::Some(s) => s.into(),
+                Option::None => 0,
```
Comment (codebestia) [inline src/account_data.cairo:501]: Looking at the implementation of how the transaction is created, the default should be 1 instead of 0 as we are using a one-index. Please update appropraitely.
Also, providing zero for the start and limit parameter should raise an error.

**Review by codebestia — CHANGES_REQUESTED** [review]
Good job so far @hakymulla 
Please address the requested changes.

**Review by codebestia — APPROVED** [review]
LGTM!
Thank you for your contribution.

**Review by codebestia — CHANGES_REQUESTED** [review]
Minor changes.

**Review by codebestia — APPROVED** [review]
LGTM!
